File: smarttrade-web/src/features/trading-journal/data/datasources/journalLocalDataSource.ts
import { db } from '@/core/persistence/db'
import type { JournalEntry } from '../../domain/entities/journalEntry'

export const journalLocalDataSource = {
  async getEntries(): Promise<JournalEntry[]> {
    return db.journalEntries.reverse().sortBy('openedAt')
  },

  async addEntry(entry: JournalEntry): Promise<void> {
    await db.journalEntries.put(entry)
  },

  async deleteEntry(id: string): Promise<void> {
    await db.journalEntries.delete(id)
  },
}

File: smarttrade_app/lib/features/trading_journal/data/datasources/journal_local_datasource.dart
import 'package:sqflite/sqflite.dart';
import 'package:smarttrade_app/core/database/app_database.dart';
import 'package:smarttrade_app/features/trading_journal/data/models/journal_entry_model.dart';
import 'package:smarttrade_app/features/trading_journal/domain/entities/journal_entry.dart';

class JournalLocalDataSource {
  const JournalLocalDataSource(this._database);

  final Database _database;

  Future<List<JournalEntry>> getEntries() async {
    final rows = await _database.query(
      AppDatabaseTables.journalEntries,
      orderBy: 'opened_at DESC',
    );
    return rows.map(JournalEntryModel.fromMap).toList();
  }

  Future<void> addEntry(JournalEntry entry) async {
    await _database.insert(
      AppDatabaseTables.journalEntries,
      JournalEntryModel.toMap(entry),
      conflictAlgorithm: ConflictAlgorithm.replace,
    );
  }

  Future<void> deleteEntry(String id) async {
    await _database.delete(
      AppDatabaseTables.journalEntries,
      where: 'id = ?',
      whereArgs: [id],
    );
  }
}
